scripts/west_commands/runners/blflash.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2021-2025 Gerson Fernando Budke <nandojve@gmail.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Bouffalo Lab flash tool (blflash) runner for serial boot ROM'''
+
+from runners.core import RunnerCaps, ZephyrBinaryRunner
+
+DEFAULT_BLFLASH_PORT = '/dev/ttyUSB0'
+DEFAULT_BLFLASH_SPEED = '2000000'
+
+
+class BlFlashBinaryRunner(ZephyrBinaryRunner):
+    '''Runner front-end for blflash.'''
+
+    def __init__(
+        self, cfg, blflash='blflash', port=DEFAULT_BLFLASH_PORT, speed=DEFAULT_BLFLASH_SPEED
+    ):
+        super().__init__(cfg)
+        self.blflash = blflash
+        self.port = port
+        self.speed = speed
+
+    @classmethod
+    def name(cls):
+        return 'blflash'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        parser.add_argument(
+            '--blflash', default='blflash', help='path to blflash, default is blflash'
+        )
+        parser.add_argument(
+            '--port',
+            default=DEFAULT_BLFLASH_PORT,
+            help='serial port to use, default is ' + str(DEFAULT_BLFLASH_PORT),
```

Comment:
Can you please use `--dev-id` instead, for consistency with other runners? This is a standard option supported in core.py -- see the nrfjprog or jlink runners for example usage; this is exposed via RunnerCaps.

Reply:
Hi,

Just to make sure we are at same page. The `blflash` tool is like `bossac` and it is not like `jlink`. It is pure software tool that uses the `bootloader` inside the ROM. So, in this case, should I still change the `port` to `dev-id` ? This means that there will be a refactor, over the time, to move from `port` / `device` to `dev-id`?

Reply:
removed from PR